### PR TITLE
Handle `asyncio.CancelledError` separately in the `LogicalMeter`

### PR DESCRIPTION
**Pull request by shsms**

### CODE REVIEW

**Review by leandro-lucarella-frequenz — DISMISSED** [review]

File: src/frequenz/sdk/timeseries/logical_meter/_logical_meter.py [hunk 1]
```diff
@@ -78,6 +78,8 @@ async def _run_formula(
             try:
                 msg = await formula.apply()
                 await sender.send(msg)
+            except asyncio.CancelledError:
+                break
```
Comment (leandro-lucarella-frequenz) [inline src/frequenz/sdk/timeseries/logical_meter/_logical_meter.py:82]: We probably want to add an error log (or `exception`), right?

File: src/frequenz/sdk/timeseries/logical_meter/_logical_meter.py [hunk 2]
```diff
@@ -78,6 +78,8 @@ async def _run_formula(
...
             except Exception as err:  # pylint: disable=broad-except
                 logger.warning("Formula application failed: %s", err)
```
Comment (ela-kotulska-frequenz) [inline src/frequenz/sdk/timeseries/logical_meter/_logical_meter.py]: Regarding: https://github.com/frequenz-floss/frequenz-sdk-python/pull/105#discussion_r1043148790

This should work ( with proper spaces :) )
```
try:  msg = await formula.apply()
except: asyncio.CancelledError: (...)
except: except Exception as err: (...)
else:  await sender.send(msg)
```

File: src/frequenz/sdk/timeseries/logical_meter/_logical_meter.py (hunk 1 above)
Comment (shsms) [reply src/frequenz/sdk/timeseries/logical_meter/_logical_meter.py:82]: done

File: src/frequenz/sdk/timeseries/logical_meter/_logical_meter.py (hunk 2 above)
Comment (shsms) [reply src/frequenz/sdk/timeseries/logical_meter/_logical_meter.py]: done

**Review by ela-kotulska-frequenz — APPROVED** [review]

File: src/frequenz/sdk/timeseries/logical_meter/_logical_meter.py (hunk 2 above)
Comment (leandro-lucarella-frequenz) [reply src/frequenz/sdk/timeseries/logical_meter/_logical_meter.py]: This is ignoring errors while sending, but that's another topic (I still have some commits to add more exceptions to the channels repo but didn't have the time to clean it up for a PR :D).

**Pull request merged**